docs/project-development-environment.md

```diff
@@ -2,7 +2,19 @@
 
 ## Local prerequisites
 
-Docker and docker-compose is required to be installed.
+| dependency | supported versions | Installation recommendation for Ubuntu 14.04 |
+| ---------- | ------------------ | ---------------------------------------------|
+| <a name='dependency_docker'>docker</a> | 1.10 (1.9 may still work, too) | Follow [the official Docker installation instructions for Linux](https://docs.docker.com/linux/step_one/) |
+| <a name='dependency_docker-compose'>docker-compose</a> | 1.6 | Install system-wide via <a href='#dependency_pip'>`pip`</a> (**not** via `pip3`! `docker-compose` is implemented in Python 2.): <pre class="highlight highlight-source-shell">sudo pip install docker-compose</pre> |
+| <a name='dependency_pg_config'>pg_config</a> (required by psycopg2) |  | <pre class="highlight highlight-source-shell">sudo apt install libpq-dev</pre> |
```

Comment:
I, too, have considered that, in two variants:

### listing reverse dependencies (i.e. the dependees), too
 I haven't yet figured out how to do that without making the table too verbose and convoluted. (Some elements are dependencies of several others. For some elements, the choice of how to install them affects what other elements need to be installed first.)

### coarsely categorize dependencies
e.g.
- **install-time dependency**, e.g. pip and &mdash;less obviously&mdash; pg_config
- **build-time dependency**, e.g. docker
- **local (test suit) runtime dependency**, e.g. pytest (would also include general runtime dependencies that have to be installed system-wide for running the test suit, like GDAL)

Many elements would belong two more than one of these categories, so separate tables would mean listing them multiple times.